### ISSUE TITLE
Bug: JavaScript Browser Documentation

### ISSUE THREAD

**Issue by jlarmstrongiv** [unum-cloud/usearch#191]
### Describe the bug

No documentation can be found for using `usearch` in the browser.

Wasmer does not have relevant documentation either.

See Discord [discussion](https://discord.com/channels/1063947616615923875/1064496121520590878/1137977581640093766)

### Steps to reproduce

Try to import `usearch` in the browser

### Expected behavior

More details or an example is provided on how to import and setup `usearch` in the browser

### USearch version

v1.1.0

### Operating System

Browser

### Hardware architecture

x86

### Which interface are you using?

Other bindings

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ashvardanian** [comment]
Thank you for acknowledging this issue, @jlarmstrongiv! We are already working with @syrusakbary and broader @wasmerio team to provide a solution 🤗

**Comment by jlarmstrongiv** [comment]
- Example of a different wasmer package https://medium.com/wasmer/wasmer-js-9a53e837b80
- More information on wasmer packages https://www.infoq.com/news/2019/10/wasmer-js-wasi-wasm-browser/

**Comment by ashvardanian** [comment]
@jlarmstrongiv, thank you fir sharing those! I will try to address those, and come back with WASM updates with a week 🤗

**Comment by jlarmstrongiv** [comment]
@ashvardanian thank you!  I’m looking forward to it 😄

**Comment by aehlke** [comment]
how has it gone? thanks.

**Comment by jlarmstrongiv** [comment]
The Wasmer JS SDK has been released this week https://wasmer.io/posts/introducing-the-wasmer-js-sdk

**Comment by ashvardanian** [comment]
@jlarmstrongiv, nice! Can it help us to run USearch in the browser?

**Comment by jlarmstrongiv** [comment]
@ashvardanian Yes, I think it would.  Especially since usearch already builds a Wasmer package

**Comment by andrewluetgers** [comment]
not sure if this helps but looking into wasmer filesystem it has an in memory fs abscraction in the browser, but it looks like they will be able to support opfs https://github.com/wasmerio/wasmer-js/issues/393

**Comment by MentalGear** [comment]
Very excited about usearch in the browser, will open up many more applications! Using OPFS for FS should work in all modern browsers!

**Comment by aehlke** [comment]
No OPFS support in WKWebView or SFSafariViewController on iOS/iPadOS/visionOS

**Comment by MentalGear** [comment]
OPFS has iOS/mac Safari support

<img width="664" alt="Screenshot 2024-01-20 at 22 27 54" src="https://github.com/unum-cloud/usearch/assets/2837147/6cd09772-7b95-456e-bee0-d2cf890bd137">

**Comment by aehlke** [comment]
I tried to find confirmation that it also works in WKWebView but couldn't find that explicitly

**Comment by MentalGear** [comment]
I don't know about WKWebView, but since it's basically safari it should. You can always test it yourself when in doubt.
RXDB has quite a great info on OPFS:

> From the beginning of 2023, the Origin Private File System API is supported by [all modern browsers](https://caniuse.com/native-filesystem-api) like Safari, Chrome, Edge and Firefox.
It is important to know that the OPFS API is only available inside of a [WebWorker](https://rxdb.info/rx-storage-worker.html). It cannot be used in the main thread, an iFrame or even a [SharedWorker](https://rxdb.info/rx-storage-shared-worker.html).

https://rxdb.info/rx-storage-opfs.html#opfs-limitations

**Comment by sridhar-mani** [comment]
Is this available to be used in browser now?

**Comment by ashvardanian** [comment]
@sridhar-mani, @MentalGear, @andrewluetgers, as I've [mentioned](https://github.com/unum-cloud/usearch/issues/246#issuecomment-2925787619) in #246, it may be wiser to adapt SimSIMD to browser environments:

> USearch isn't the best fit for the browser. It's designed for approximate search over extensive collections. Browser environments are generally limited to 4GB, and even that is generous.
>
> Most browser-like use cases can be better implemented via a fast vector-math library, like the underlying SimSIMD, which has a [WASM-related draft](https://github.com/ashvardanian/SimSIMD/pull/242/files). I'm not sure how one would use it, as it only focuses on the core C library rather than the JavaScript bindings. It also doesn't cover the packaging or hosting part in the CI, just the build.

It would be great if someone more familiar with the complex world of modern browser environments could take a look and try to finish that line of work. Maybe it's just a matter of a couple of new entries in binding.gyp and release.yml 🤷
